starfish/test/recipe/test_runnable.py

```diff
@@ -0,0 +1,170 @@
+import numpy as np
+import pytest
+
+from starfish import Codebook, ImageStack, IntensityTable
+from starfish.recipe.runnable import Runnable
+from .fakefilter import FakeFilterAlgorithm  # noqa: F401
+
+
+def test_get_file_input_types():
+    """Verify that we can map from inputs that are meant to be filepaths to the expected object
+    type.
+
+    If you're looking at this because this test is failing, ask yourself if you've changed
+    PerRoundMaxChannelDecoder (either the name of the algorithm or its parameters).  If so, then
+    you can update the expected outcome of the test.  If you did not, then it's probably a
+    legitimate failure.
+    """
+    runnable = Runnable(
+        "decode", "PerRoundMaxChannelDecoder",
+        "should_be_intensity_table", "should_be_codebook")
+    file_input_types = runnable.file_input_types
+    assert len(file_input_types) == 2
+    assert file_input_types["should_be_intensity_table"] == IntensityTable
+    assert file_input_types["should_be_codebook"] == Codebook
+
+
+def test_get_file_input_types_from_runnable():
+    """Verify that we can map from inputs, some of which are meant to be filepaths, to the expected
+     object type.
+
+    If you're looking at this because this test is failing, ask yourself if you've changed
+    BlobDetector or PerRoundMaxChannelDecoder (either the name of the algorithm or its parameters).
+    If so, then you can update the expected outcome of the test.  If you did not, then it's probably
+    a legitimate failure.
+    """
+    intensity_table_runnable = Runnable(
+        "detect_spots", "BlobDetector",
+        "should_be_image_stack",
+        min_sigma=1,
+        max_sigma=1,
+        num_sigma=1,
+        threshold=1,
+        overlap=1,
+    )
+    runnable = Runnable(
+        "decode", "PerRoundMaxChannelDecoder",
+        intensity_table_runnable, "should_be_codebook")
+
+    file_input_types = intensity_table_runnable.file_input_types
+    assert len(file_input_types) == 1
+    assert file_input_types["should_be_image_stack"] == ImageStack
+
+    file_input_types = runnable.file_input_types
+    assert len(file_input_types) == 1
+    assert file_input_types["should_be_codebook"] == Codebook
+
+
+def test_run():
+    """Verify that we can run a single runnable and get its result.
+    """
+    filter_runnable = Runnable(
+        "filter", "FakeFilterAlgorithm",
```

Comment:
why not just use a real filter class here?

Reply:
It's easier b/c getting the expected results are easier. :)